python/sanlock.c

```diff
@@ -1611,24 +1611,27 @@ with -EBUSY error in this case.\n\
 static PyObject *
 py_set_event(PyObject *self __unused, PyObject *args, PyObject *keywds)
 {
-    const char *lockspace;
+    PyObject *lockspace = NULL;
     struct sanlk_host_event he = {0};
     uint32_t flags = 0;
     int rv;
 
     static char *kwlist[] = {"lockspace", "host_id", "generation", "event",
                              "data", "flags", NULL};
 
-    if (!PyArg_ParseTupleAndKeywords(args, keywds, "sKKK|KI", kwlist,
-        &lockspace, &he.host_id, &he.generation, &he.event, &he.data,
-        &flags)) {
+    if (!PyArg_ParseTupleAndKeywords(args, keywds, "O&KKK|KI", kwlist,
+        convert_to_pybytes, &lockspace, &he.host_id, &he.generation, &he.event,
+        &he.data, &flags)) {
+        Py_XDECREF(lockspace);
         return NULL;
     }
 
     Py_BEGIN_ALLOW_THREADS
-    rv = sanlock_set_event(lockspace, &he, flags);
+    rv = sanlock_set_event(PyBytes_AsString(lockspace), &he, flags);
     Py_END_ALLOW_THREADS
 
+    Py_DECREF(lockspace);
+
     if (rv < 0) {
         __set_exception(rv, "Unable to set event");
         return NULL;
```

Comment:
Should use same finally pattern as previous patch.

Reply:
See prev comment